test-utils/README.md

```diff
@@ -0,0 +1,42 @@
+# `libcrux-test-utils`
+
+This crate contains tools for testing, tracing, profiling and benchmarking.
+
+## Tracing (`tracing`)
+
+This module provides tools for annotating functions such that calls are traced.
```

Comment:
What's the story for putting this into production code?
We need a way to enable this, while making it no-ops in regular builds.